src/docs/go-grpc/sections/03-operationalize/01-renewal/01-builtin.mdx

```diff
@@ -0,0 +1,93 @@
+The above example will work as long as the certificates do not expire. By
+default, `step-ca` certificates expire after 24 hours, so to go above that, we
+need a way to renew the server certificates automatically.
+
+One of the best options is to use the ACME protocol, and instead of using `step`
+to create the certificate, we will use the package
+[acme/autocert](https://pkg.go.dev/golang.org/x/crypto/acme/autocert) that
+provides an easy way to support ACME on Go gRPC (and HTTP) servers. It provides
+automatic access to certificates from Let's Encrypt or a private ACME server
+like `step-ca`.
+
+Using `acme/autocert` the `autocert.Manager` will now take care of creating a
+certificate signing request (CSR) and requesting a certificate using the ACME
+protocol. It will also take care of renewing the certificate, to do this we need
+to make sure to set a proper `RenewBefore` according to the lifetime of your
+certificates. By default, `RenewBefore` is set to 30 days before the expiration,
+but if our certificate lifetime is 24 hours the renewal will kick on every
+connection, to avoid this we can for example renew 8 hours before the expiration
+setting `RenewBefore` to `8 * time.Hour`.
```

Comment:
```suggestion
certificates. By default, `RenewBefore` is set to 30 days before the expiration.
Since our certificate lifetime is 24 hours, the `autocert.Manager` will attempt renewal _on every connection_ unless we adjust this setting.
We recommend attempting renewal when a certificate has exceeded two-thirds of its lifetime.
To attempt renewal 8 hours before the certificate expires, set `RenewBefore` to `8 * time.Hour`.
```